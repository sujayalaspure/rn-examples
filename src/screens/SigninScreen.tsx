import { Text, KeyboardAvoidingView, StyleSheet } from 'react-native';
import React, { useState } from 'react';
import { Button, ErrorText, Input } from '../components/Form';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    justifyContent: 'center',
    paddingHorizontal: 40
  },
  headerText: {
    color: '#353031',
    fontWeight: 'bold',
    fontSize: 34,
    marginBottom: 10
  }
});

const SigninScreen = () => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [errorMsg, setErrorMsg] = useState('');

  const submit = () => {
    if (!username || !password) {
      setErrorMsg('Please enter username and password');
      return;
    }
    setErrorMsg('');
    console.log('submit', { username, password });
  };

  return (
    <KeyboardAvoidingView style={styles.container} behavior="position">
      <Text style={styles.headerText}>Login</Text>
      <Input
        label="Username"
        placeholder="Username"
        onChangeText={text => setUsername(text)}
        error={errorMsg}
        testID="SignIn.usernameInput"
      />
      <Input
        label="Password"
        placeholder="***"
        secureTextEntry
        onChangeText={text => setPassword(text)}
        error={errorMsg}
        testID="SignIn.passwordInput"
      />
      <ErrorText messages={[errorMsg]} />
      <Button testID="SignIn.Button" text="Login" onPress={submit} />
    </KeyboardAvoidingView>
  );
};

export default SigninScreen;
