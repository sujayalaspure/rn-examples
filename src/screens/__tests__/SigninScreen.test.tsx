import React from 'react';
import { render } from '@testing-library/react-native';

import SigninScreen from '../SigninScreen';
// eslint-disable-next-line @typescript-eslint/no-var-requires
require('react-native-reanimated/lib/commonjs/reanimated2/jestUtils').setUpTests();

// eslint-disable-next-line global-require
jest.mock('react-native-reanimated', () => require('react-native-reanimated/mock'));

describe('SigninScreen', () => {
  it('matches the snapshot', () => {
    const { toJSON } = render(<SigninScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders username field', () => {
    const { getByTestId } = render(<SigninScreen />);
    const myComponent = getByTestId('SignIn.usernameInput');
    expect(myComponent).toBeDefined();
    expect(myComponent.props.placeholder).toEqual('Username');
  });
  it('renders password field', () => {
    const { getByTestId } = render(<SigninScreen />);
    const myComponent = getByTestId('SignIn.passwordInput');
    expect(myComponent).toBeDefined();
    expect(myComponent.props.placeholder).toEqual('***');
  });
  it('renders login button', () => {
    const { getByTestId } = render(<SigninScreen />);
    const myComponent = getByTestId('SignIn.Button');
    expect(myComponent).toBeDefined();
  });
});
