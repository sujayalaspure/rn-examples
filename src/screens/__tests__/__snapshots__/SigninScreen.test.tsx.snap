// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SigninScreen matches the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 40,
    }
  }
>
  <View
    style={
      {
        "bottom": 0,
      }
    }
  >
    <Text
      style={
        {
          "color": "#353031",
          "fontSize": 34,
          "fontWeight": "bold",
          "marginBottom": 10,
        }
      }
    >
      Login
    </Text>
    <View
      style={
        [
          {
            "backgroundColor": "#f4f6f8",
            "borderColor": "#f4f6f8",
            "borderRadius": 10,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
          },
          null,
        ]
      }
    >
      <Text
        style={
          {
            "color": "#b4b6b8",
            "fontSize": 10,
          }
        }
      >
        Username
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Username"
          style={
            {
              "color": "#353031",
              "flex": 1,
              "fontSize": 14,
              "fontWeight": "bold",
              "marginRight": 10,
              "marginTop": 3,
            }
          }
          testID="SignIn.usernameInput"
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#f4f6f8",
            "borderColor": "#f4f6f8",
            "borderRadius": 10,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
          },
          null,
        ]
      }
    >
      <Text
        style={
          {
            "color": "#b4b6b8",
            "fontSize": 10,
          }
        }
      >
        Password
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="***"
          secureTextEntry={true}
          style={
            {
              "color": "#353031",
              "flex": 1,
              "fontSize": 14,
              "fontWeight": "bold",
              "marginRight": 10,
              "marginTop": 3,
            }
          }
          testID="SignIn.passwordInput"
        />
      </View>
    </View>
    <View
      style={
        {
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#cc0011",
            "fontSize": 14,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#9374b7",
          "borderRadius": 10,
          "justifyContent": "center",
          "marginTop": 10,
          "opacity": 1,
          "paddingVertical": 16,
        }
      }
      testID="SignIn.Button"
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
